operations/bills.md

```diff
@@ -183,6 +183,159 @@ Extent of data to be returned. E.g. it is possible to specify that together with
 | `Bills` | array of [Bill](bills.md#bill) | required | The filtered bills. |
 | `Cursor` | string | optional | Unique identifier of the last and hence oldest bill returned. This can be used in `Limitation` in a subsequent request to fetch the next batch of older bills. |
 
+## Update Bills
+
+Updates one or more existing bills in the system. Closed bills cannot be updated. Note this operation supports [Portfolio Access Tokens](../guidelines/multi-property.md).
```

Comment:
Except... I was looking at the operation, trying to understand what it does, and I'm not sure "Updates the bill" is a very good description in any case. The parameters are `OwnerId` and `AssociatedAccountId`, so I guess this operation is to change the assignment of the bill, i.e. which accounts the bill is assigned to? Have I understood this correctly? If so, I would consider changing the name of the operation, or at least changing the description.

I also think it would be good to update the descriptions for `OwnerId` and `AssociatedAccountId` to better explain what they are. I'm assuming "owner" is the owning Account and there may be a link to a second associated Account as well. Is that for example if it's a business stay so one would be the guest and the other the invoiced company?

It's always best to assume the API user is not an expert on Mews terminology and data structure, and I try to put myself in the user's shoes (which isn't that hard because I'm no expert!).

Thanks, I would appreciate a review of this.

Reply:
@jnv , just a quick reminder to add your comment when you get a chance 🙂.

Reply:
Ok thanks @ShadiWa and @jnv I understand, the updated description is clearer.

I think we still need to be careful about parameters. For example, the two update parameters are `OwnerId` and `AssociatedAccountId`, but in the Bill object that's returned, the fields are `AccountId` (not `OwnerId`) and `AssociatedAccountIds` (plural, not singular). This creates questions that need to be resolved, so I would recommend:

1. Either change `OwnerId` to `AccountId` or be very clear if they are the same thing. I can't see any reason _not_ to use the same name as the field we will be updating, so I think it should be changed to `AccountId`; I would also make sure the description aligns with the description of `AccountId` in `Bill`, so terminology is consistent, e.g. do we use the term "owned" or "assigned"?
2.  We need to explain which of the `AssociatedAccountIds` we are updating, as they are plural (an array). I believe it was made plural for future expansion, but that doesn't change the fact; again we should make sure terminology is aligned.
3. If I am right that `AssociatedAccountIds` is plural for future expansion reasons, we should add a note to the description in `Bills` to make that clear.

Reply:
@MikeAdamsMews You are 100% right.

1. This is my mistake when I was reviewing the initial design of the endpoint, I missed that we call owner AccountId. I am preparing a PR which will change OwnerId to AccountId, along with documentation suggestions.
2. Yeah, I am torn on this one. Accounts team introduced AssociatedAccountIds as array with limit of a single account. We don't support multiple associated account IDs but they wanted to future-proof it. In `/add` endpoint, we support `AssociatedAccountId` (singular) but adding support for multiple accounts there will be simple backward compatible change. Not in the case of update, though.
3. I think I will change `AssociatedAccountId` to an array.